kernel/device.c

```diff
@@ -65,10 +65,6 @@ struct device *device_get_binding(const char *name)
 			continue;
 		}
 
-		if (name == info->config->name) {
-			return info;
-		}
-
```

Comment:
The optimization as it is isn't helping much: the comparison will be false for all strings with the exception of the very last one (in the worst case scenario).

A way to "fix" this would be to split the loop in two: one that only compares pointers, and one that only compares with strcmp(). In the majority of cases, we'll be comparing just the pointers in ROM, and the "fallback" with strcmp() will never be executed, but it needs to be there.